app/controllers/users/settings/teams_controller.rb

```diff
@@ -147,7 +147,7 @@ def load_user
 
       def load_team
         @team = Team.find_by_id(params[:id])
-        render_403 unless is_admin_of_team(@team)
+        render_403 unless can_read_team?(@team)
```

Comment:
I would remove this check from `load_team` and add separate method `check_update_team_permission`, where you would check `can_update_team?(@team)`

Reply:
but there is a banana. Until now we didn't show this controller actions if the user was not an admin

Reply:
Should I make a new permission check like can_manage_team?

Reply:
Just use then `can_update_team?(@team)`